agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,54 +33,122 @@
    */ 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
Would it simplify this change if you added a `Collection<Strings> attendees` parameter to this function (i.e. move the existing code in this function to a new function with 3 parameters instead of 2)? Then from this method you could call into that function with both sets of attendees (optional+required and required-only).

Reply:
@ykjchen  I see what you're saying. My original reasoning was that this method only requires creating the minutes array once and processing it once. The code is less simple, but more efficient. Do you think it's worth the trade off in efficiency to change it? If so, I'm happy to make the change.

Reply:
Thanks for the elaboration. I think I'll try and make the change. There will be a slight decrease in efficiency; however, I think it should be pretty minimal to just add an extra sweep of the array.

Reply:
I decided to keep the current algorithm, as it would require a total change of the algorithm in this class to make this change, and I think the current one is slightly more efficient. However, I definitely agree it's less understandable, which is why I chose to adopt the enum, and add more comments to clarify my algorithm. Additionally, this algorithm served as a sort of stepping stone to the last algorithm I ended up with for the challenge, so that's also a part of why I wanted to preserve it.

Reply:
Done